tests/integration/adapter-tests.js

```diff
@@ -57,6 +57,90 @@ export default function adapterTests() {
     expect(await persister.find(recordingId)).to.be.null;
   });
 
+  // TODO
+  it('re-records on expired recording if recordIfExpired is true', async function() {
+    this.timeout(10000);
+
+    const url = '/api';
+    const { persister, server } = this.polly;
+    let pollyRequest;
+
+    server.get(url).on('request', req => {
+      pollyRequest = req;
+    });
+
+    this.polly.configure({
+      recordIfExpired: true,
+      expiresIn: '1ms',
+      matchRequestsBy: {
+        order: false
+      }
+    });
+
+    await this.relativeFetch(url);
+    await persister.persist();
+    const { startedDateTime: prevDateTime } = await persister.findEntry(
+      pollyRequest
+    );
+
+    await new Promise(r => setTimeout(r, 10));
+
+    await this.relativeFetch(url);
+    await persister.persist();
+    const { startedDateTime: newDateTime } = await persister.findEntry(
+      pollyRequest
+    );
+
+    expect(new Date(prevDateTime)).to.be.lessThan(new Date(newDateTime));
+  });
```

Comment:
we're having a pretty hard time figuring out the best way to test expiration. we'd thought that persisting in between checks in the test would clear the persister cache and allow us to test that the file had a timestamp that was then "expired" given a very low expiration time; any help y'all can provide would be really helpful

Reply:
Ok, took me a little bit of debugging but I figured out what was going on. The persister does reset the cache but the `persister` instance you're holding on to is an old one (its a pretty troll behavior that we can get around by doing diffing of the new and old config to determine what needs to be created). 

A new persister instance was created after you did `this.polly.configure()`. The following should fix it: 

```js
    this.polly.configure({
      recordIfExpired: true,
      expiresIn: '1ms',
      matchRequestsBy: {
        order: false
      }
    });

    const url = '/api';
    const { persister, server } = this.polly;
```

In terms of testing strategy, I think that testing against the startedDateTime should be sufficient but you can also tap into the `beforePersist` event to test that it was being triggered twice.

Reply:
Just as a note here, you're going to have to manually delete the recording after every test since its not something we want to be persisted on the fs. You can use `await persister.delete(recordingId)` to do so. Might be worth moving all theses expiration tests to be under their own `describe` block.

Reply:
yep, we were thinking that would be needed. We're thinking a better solution for these tests longer-term would be to implement an in-memory persister to use for some tests that test the internals like this; maybe we'd contribute that in the future if there are other uses of an in-memory persister

Reply:
Interesting! An in-memory persister can be quite useful just for testing purposes. I'll open a new issue for it to better track this and see if I can come up with more uses-cases.